kernel/fde/optee/fde_key_handler_ta_type.h

```diff
@@ -0,0 +1,95 @@
+/* SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright Canonical Ltd.
+ */
+
+/*
+ * This file lives in https://github.com/canonical/optee-uc-fde and an identical
+ * copy is kept in https://github.com/canonica/snapd. Updating the original file
+ * requires that the copy in the snapd repo is also updated.
+ */
+
+#ifndef FDE_KEY_HANDLER_TA_TYPE_H_
+#define FDE_KEY_HANDLER_TA_TYPE_H_
+
+// fd1b2a86-3668-11eb-adc1-0242ac120002
+#define FDE_KEY_HANDLER_UUID_ID {0xfd1b2a86, 0x3668, 0x11eb, \
+    { \
+        0xad, 0xc1, 0x02, 0x42, 0xac, 0x12, 0x00, 0x02 \
+    } \
+}
+
+#define MAX_BUF_SIZE    512
+#define HANDLE_SIZE     65  // (version)(1), IV(16), NONCE(32), TAG(16)
+
+/* Define the command index in this TA */
+
+/*
+ * TA_CMD_KEY_ENCRYPT have 3 parameters
+ * Encrypts passed key with derived key
+ * Key handle is randomply generated at key derivation
+ * - TEE_PARAM_TYPE_MEMREF_INPUT
+ *    params[0].memref.buffer: plain key buffer
+ *    params[0].memref.size: length of the buffer
+ * - TEE_PARAM_TYPE_MEMREF_OUTPUT
+ *    params[1].memref.buffer: returned key handle
+ *    params[1].memref.size: length of the buffer
+ * - TEE_PARAM_TYPE_MEMREF_OUTPUT
+ *    params[2].memref.buffer: returned encrypted key
+ *    params[2].memref.size: length of the buffer
+ */
+#define TA_CMD_KEY_ENCRYPT            1U
```

Comment:
There seems to be something missing in API for authenticated encryption. Which means the secure side code does not support that. Though I am not totally sure this is needed here. But it would be interesting to have @chrisccoulson 's opinion on that.

Reply:
The version is for changing this API. So that is OK.